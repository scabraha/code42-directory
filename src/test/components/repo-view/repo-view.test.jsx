import React from 'react';
import renderer from 'react-test-renderer';
import RepoView from '../../../components/repo-view/repo-view';

describe('Repo View', () => {
  it('Renders the view without repo details and loading status and ensures it renders correctly', () => {
    const tree = renderer.create(
      <RepoView repoRequestStatus="LOADING"  repos={[]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view without repo details and failure status and ensures it renders correctly', () => {
    const tree = renderer.create(
      <RepoView repoRequestStatus="FAILURE"  repos={[]} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders the view with repo details and ensures it renders correctly', () => {
    const tree = renderer.create(
      <RepoView 
        repoRequestStatus="SUCCESS"  
        repos={[
          {
            html_url: 'url1',
            name: 'repo1',
            language: 'language1',
            description: 'description1'
          },
          {
            html_url: 'url2',
            name: 'repo2',
            language: 'language2',
            description: 'description2'
          },
          {
            html_url: 'url3',
            name: 'repo3',
            language: 'language3',
            description: 'description3'
          },
        ]} 
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

