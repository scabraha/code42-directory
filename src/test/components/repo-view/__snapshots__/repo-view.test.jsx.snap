// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repo View Renders the view with repo details and ensures it renders correctly 1`] = `
<ul
  className="repo-list"
>
  <li
    className="repo-list-item"
  >
    <div
      className="md-title"
    >
      <a
        href="url1"
      >
        repo1
      </a>
    </div>
    <div
      className="md-body-2"
    >
      description1
    </div>
    <div
      className="md-body-1"
    >
      language1
    </div>
  </li>
  <li
    className="repo-list-item"
  >
    <div
      className="md-title"
    >
      <a
        href="url2"
      >
        repo2
      </a>
    </div>
    <div
      className="md-body-2"
    >
      description2
    </div>
    <div
      className="md-body-1"
    >
      language2
    </div>
  </li>
  <li
    className="repo-list-item"
  >
    <div
      className="md-title"
    >
      <a
        href="url3"
      >
        repo3
      </a>
    </div>
    <div
      className="md-body-2"
    >
      description3
    </div>
    <div
      className="md-body-1"
    >
      language3
    </div>
  </li>
</ul>
`;

exports[`Repo View Renders the view without repo details and failure status and ensures it renders correctly 1`] = `
<div
  className="md-body-2"
>
  Failed to retrieve repositories
</div>
`;

exports[`Repo View Renders the view without repo details and loading status and ensures it renders correctly 1`] = `
<i
  className="loading-spinner material-icons"
>
  î¡£
</i>
`;
